### PR TITLE
feat(link): allow usage of icon - INNO-1176

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

- Allows user to add icons before or after a link
- Add EU links
- Cleanup markup of links

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 805a04ece79c13d76c766f0f1f2751f496baafd0

https://deploy-preview-812--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/packages/link/Link.jsx [hunk 1]
```diff
@@ -1,44 +1,69 @@
...
+  icon,
+  iconPosition,
+  className,
   ...props
 }) => {
-  let classes = 'ecl-link';
-  if (variant) classes += ` ecl-link--${variant}`;
-  if (extraClasses) classes += ` ${extraClasses}`;
+  const classNames = classnames(className, {
+    'ecl-link': true,
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/link/Link.jsx]: `classnames(className, { 'ecl-link': true, ` -> `classnames(className, 'ecl-link', { `

File: src/systems/ec/implementations/react/packages/link/Link.jsx [hunk 2]
```diff
@@ -1,44 +1,69 @@
...
+    [`ecl-link--${variant}`]: variant,
+    [`ecl-link--icon-${iconPosition}`]: icon && icon.icon,
+  });
+
+  let link = '';
+  if (icon && icon.icon) {
+    if (iconPosition === 'before') {
+      link = (
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/link/Link.jsx]: why not `return` directly? (no need for `link` variable)

File: src/systems/ec/implementations/react/packages/link/Link.jsx [hunk 3]
```diff
@@ -1,44 +1,69 @@
...
+        <a {...props} href={href} className={classNames}>
+          <Icon {...icon} className="ecl-link__icon" />
```
Comment (yhuard) [inline src/systems/ec/implementations/react/packages/link/Link.jsx]: `icon.className` is lost on the way. 

-> `<Icon {...icon} className={classnames(icon.className, 'ecl-link__icon')} />`

File: src/systems/ec/specs/components/link/docs/code.mdx [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
+import demoContentDefault from '@ecl/ec-specs-link/demo/data--default';
+import demoContentStandalone from '@ecl/ec-specs-link/demo/data--standalone';
+import demoContentIcon from '@ecl/ec-specs-link/demo/data--icon';
+
+## Default link
+
 <Playground playgroundLink="/storybook/ec/index.html?selectedKind=Link&selectedStory=default&stories=0">
   <Link
+    {...demoContentDefault}
     variant="default"
```
Comment (yhuard) [inline src/systems/ec/specs/components/link/docs/code.mdx]: why not put the `variant` in the demo content?

File: src/systems/generic/implementations/vanilla/generic-component-link/_mixins.scss [hunk 5]
```diff
@@ -4,7 +4,6 @@
   $link-color-focus,
   $link-color-visited
 ) {
-  box-sizing: border-box;
   color: $link-color;
   font-family: $ecl-font-family;
```
Comment (yhuard) [inline src/systems/generic/implementations/vanilla/generic-component-link/_mixins.scss]: Font family should be inherited unless it's a standalone link

File: src/systems/eu/specs/components/link/demo/data--icon.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+// Simple content for demo
+module.exports = {
+  label: 'Link with icon',
+  href: '/example#link-icon',
+  icon: {
+    icon: 'Icon_Corner-arrow-right',
+    size: 'xs',
```
Comment (yhuard) [inline src/systems/eu/specs/components/link/demo/data--icon.js]: By default, the dimension should be 1em x 1em

**Review by yhuard — DISMISSED** [review]

**Comment by emeryro** [comment]
Modifications done, ready for a second round

**Review by degliwe — COMMENTED** [review]
spacing between icon and link is bigger than spacing between words.

**Comment by emeryro** [comment]
Spacing updated

**Review by degliwe — APPROVED** [review]

**Pull request merged**